website/source/docs/providers/consul/r/prepared_query.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: "consul"
+page_title: "Consul: consul_prepared_query"
+sidebar_current: "docs-consul-resource-prepared-query"
+description: |-
+  Allows Terraform to manage a Consul prepared query
+---
+
+# consul\_prepared\_query
+
+Allows Terraform to manage a Consul prepared query.
+
+Managing prepared queries is done using Consul's REST API. This resource is
+useful to provide a consistent and declarative way of managing prepared
+queries in your Consul cluster using Terraform.
+
+## Example Usage
+
+```
+resource "consul_prepared_query" "service-near-self" {
+    datacenter = "nyc1"
+    token = "abcd"
+    stored_token = "wxyz"
+    name = ""
+    service = "$${match(1)}"
+    only_passing = true
+    near = "_agent"
+
+    failover {
+        nearest_n = 3
+        datacenters = ["dc2", "dc3", "dc4"]
+    }
+
+    dns {
+        ttl = "5m"
+    }
+
+    template {
```

Comment:
Since the `service` uses this regex, I'd put the `template` definition higher up.